### PR TITLE
fix: use go 1.19 only for integration tests on Darwin

### LINKED ISSUE #19802: Integration & Contract Tests are failing on MacOS after go upgrade 1.20
Integration & Contract Tests are failing on MacOS after upgrading to Go version 1.20. This issue is only reproducible on MacOS; tests work fine on NixOS.

The issue was introduced by: https://github.com/status-im/status-mobile/pull/19564

Error:
```zsh
/private/tmp/tmp-status-mobile-6d6766dce1/nix-shell-92776-0/rc: line 3: 93011 Abort trap: 6           node --require ./test-resources/override.js "$SHADOW_OUTPUT_TO"
make: *** [_test-clojure] Error 134
```

It would be nice to implement a MacOS CI pipeline to catch such issues in the future.

cc: @siddarthkay

### PR DESCRIPTION
**Pull request by siddarthkay**
fixes #19802 

## Summary

Contract Tests are failing with missing symbols runtime error on `MacOS` after `golang` version was upgraded to 1.20
A simple hack fix such a case is to use `go 1.19` only to build the status-go library that is used specifically for these integration tests.
Its ugly but it works and unblocks devs from running tests locally.

## Review Notes
- execute `make test-contract` on MacOS and it should not fail.

## Testing notes
not needed since this impacts only the integration tests.

## Platforms
- macOS

status: ready

### CODE REVIEW

File: nix/overlay.nix [hunk 1]
```diff
@@ -60,6 +60,9 @@ in {
     versions = ["15.1"];
   };
   go = super.go_1_20;
+  # https://github.com/status-im/status-mobile/issues/19802
+  # only for Darwin to fix Integration Tests failing with missing symbols on go 1.20
+  buildGoPackageIntegrationTest = if stdenv.isDarwin then super.buildGo119Package else super.buildGo120Package;
```
Comment (jakubgs) [inline nix/overlay.nix]: It doesn't make any sense to put this in `nix/overlay.nix`. This is a place that should affect things globally.

Comment (siddarthkay) [reply nix/overlay.nix]: Moving this logic over to status-go-library derivation

File: nix/overlay.nix [hunk 2]
```diff
@@ -61,6 +61,7 @@ in {
   };
   go = super.go_1_20;
   clang = super.clang_15;
+  buildGo19Package = super.buildGo19Package;
```
Comment (jakubgs) [inline nix/overlay.nix]: This is unnecessary.

File: nix/status-go/library/default.nix [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
-, meta, source}:
-
-buildGoPackage {
+{ stdenv, meta, source, buildGo119Package, buildGo120Package }:
+let
+# https://github.com/status-im/status-mobile/issues/19802
+# only for Darwin to fix Integration Tests failing with missing symbols on go 1.20
+buildGoPackageIntegrationTest = if stdenv.isDarwin then buildGo119Package else buildGo120Package;
+in
+buildGoPackageIntegrationTest {
```
Comment (jakubgs) [inline nix/status-go/library/default.nix]:
```suggestion
let
  # https://github.com/status-im/status-mobile/issues/19802
  # only for Darwin to fix Integration Tests failing with missing symbols on go 1.20
  buildGoPackageIntegrationTest = if stdenv.isDarwin then buildGo119Package else buildGo120Package;
in buildGoPackageIntegrationTest {
```

**Review by jakubgs — APPROVED** [review]

File: nix/overlay.nix (hunk 2 above)
Comment (siddarthkay) [reply nix/overlay.nix]: indeed.
Removed, Thanks!

File: nix/status-go/library/default.nix (hunk 3 above)
Comment (siddarthkay) [reply nix/status-go/library/default.nix]: fixed in latest commit.

**Review by cammellos — APPROVED** [review]

**Pull request merged**